src/module.c

```diff
@@ -11426,71 +11453,81 @@ int RM_RegisterStringConfig(RedisModuleCtx *ctx, const char *name, RedisModuleSt
         enum_vals[i].val = int_values[i];
     }
     enum_vals[num_enum_vals].name = NULL;
     enum_vals[num_enum_vals].val = 0;
-    addModuleConfigtoList(module, new_config);
+    listAddNodeTail(module->module_configs, new_config);
     flags = maskModuleConfigFlags(flags);
     addModuleEnumConfig(module->name, name, flags, new_config, default_val, enum_vals);
+    module->has_configs = 1;
     return REDISMODULE_OK;
 }
+
 /*
  Create an integer config that server clients can interact with via the 
  * `CONFIG SET`, `CONFIG GET`, and `CONFIG REWRITE` commands. See 
  * RedisModule_RegisterStringConfig for detailed information about configs. */
 int RM_RegisterNumericConfig(RedisModuleCtx *ctx, const char *name, long long default_val, unsigned int flags, long long min, long long max, RedisModuleConfigGetNumericFunc getfn, RedisModuleConfigSetNumericFunc setfn, RedisModuleConfigApplyFunc applyfn, void *privdata) {
     RedisModule *module = ctx->module;
-    if (moduleConfigValidityCheck(module, name, flags, NUMERIC_CONFIG)) {
+    sds config_name = sdsnew(name);
+    if (moduleConfigValidityCheck(module, config_name, flags, NUMERIC_CONFIG)) {
         return REDISMODULE_ERR;
     }
-    ModuleConfig *new_config = createModuleConfig(sdsnew(name), applyfn, privdata, module);
-    addDefaultAndCallbacksToConfig(new_config, getfn, setfn, numeric);
-    addModuleConfigtoList(module, new_config);
+    ModuleConfig *new_config = createModuleConfig(config_name, applyfn, privdata, module);
+    new_config->get_fn.get_numeric = getfn;
+    new_config->set_fn.set_numeric = setfn;
+    listAddNodeTail(module->module_configs, new_config);
     unsigned int numeric_flags = maskModuleNumericConfigFlags(flags);
     flags = maskModuleConfigFlags(flags);
     addModuleNumericConfig(module->name, name, flags, new_config, default_val, numeric_flags, min, max);
+    module->has_configs = 1;
     return REDISMODULE_OK;
 }
 
 /* Applies all pending configurations on the module load. This should be called
- * after all of the configurations have been registered for the module. This API
- * needs to be called when configurations are provided in either `MODULE LOADEX`
+ * after all of the configurations have been registered for the module inside of RedisModule_OnLoad.
+ * This API needs to be called when configurations are provided in either `MODULE LOADEX`
  * or provided as startup arguments. */
 int RM_LoadConfigs(RedisModuleCtx *ctx) {
     if (!ctx || !ctx->module) {
         return REDISMODULE_ERR;
     }
     RedisModule *module = ctx->module;
-    /* Load configs from conf file */
-    if (loadModuleConfigs(module)) {
-        return REDISMODULE_ERR;
+    ctx->load_configs_called = 1;
+    /* Load configs from conf file or arguments from loadex */
+    if (ctx->is_loadex) {
+        if (loadModuleConfigs(module, 1)) return REDISMODULE_ERR;
+    } else {
+        if (loadModuleConfigs(module, 0)) return REDISMODULE_ERR;
     }
```

Comment:
```suggestion
    if (loadModuleConfigs(module, ctx->is_loadex)) return REDISMODULE_ERR;
```
all of that won't actually be needed since i think loadModuleConfigs doesn't need to take that argument, and instead the validation that there are no configs unhandled should be in the LOADEX command itself.
actually it also means that we can remove `is_loadex` from the module context (triple win).

now if only we can find a way to also find a way to get rid of the excess `load_configs_called` member (present in all contexts, even ones not used for the module loading), that would be great.